test/evalUtils.js

```diff
@@ -582,6 +626,83 @@ describe('test/utils/evalUtils.js >', (done) => {
         done();
       }
     });
+
+    describe('logging >', () => {
+      const methods = ['log', 'info', 'error', 'warn'];
+      let spies = {};
+
+      before(() => {
+        methods.forEach((key) => spies[key] = sinon.spy(console, key));
+      });
+
+      afterEach(() => {
+        methods.forEach((key) => spies[key].reset());
+      });
+
+      after(() => {
+        methods.forEach((key) => spies[key].restore());
+      });
+
+      it('logging not allowed by default', (done) => {
+        const str = `console.log('1');console.info('2');console.error('3');` +
+          `console.warn('4')`;
+        try {
+          eu.safeEval(str);
+          expect(spies.log.called).to.be.false;
+          expect(spies.info.called).to.be.false;
+          expect(spies.error.called).to.be.false;
+          expect(spies.warn.called).to.be.false;
+          done();
+        } catch (err) {
+          done(err);
+        }
+      });
+
+      it('logging ok with flag', (done) => {
+        const str = `console.log('1');console.info('2');console.error('3');` +
+          `console.warn('4');console.log('1');console.info('2');` +
+          `console.error('3');console.warn('4');`;
+        try {
+          eu.safeEval(str, {}, true);
+          expect(spies.log.callCount).to.equal(2);
+          expect(spies.info.callCount).to.equal(2);
+          expect(spies.error.callCount).to.equal(2);
+          expect(spies.warn.callCount).to.equal(2);
+          expect(spies.log.calledWith('1')).to.be.true;
+          expect(spies.info.calledWith('2')).to.be.true;
+          expect(spies.error.calledWith('3')).to.be.true;
+          expect(spies.warn.calledWith('4')).to.be.true;
+          done();
+        } catch (err) {
+          done(err);
+        }
+      });
+
+      it('logging ok even if error', (done) => {
+        const str = `console.log('1');console.info('2');console.error('3');` +
+          `console.warn('4');aaa.bbb();console.log('1');console.info('2');` +
+          `console.error('3');console.warn('4');`;
+        try {
+          eu.safeEval(str, {}, true);
+          done('Expecting FunctionBodyError here');
+        } catch (err) {
+          if (err.name === 'FunctionBodyError') {
+            expect(spies.log.callCount).to.equal(1);
+            expect(spies.info.callCount).to.equal(1);
+            expect(spies.error.callCount).to.equal(1);
+            expect(spies.warn.callCount).to.equal(1);
+            expect(spies.log.calledWith('1')).to.be.true;
+            expect(spies.info.calledWith('2')).to.be.true;
+            expect(spies.error.calledWith('3')).to.be.true;
+            expect(spies.warn.calledWith('4')).to.be.true;
+            done();
+          } else {
+            done('Expecting FunctionBodyError here');
```

Comment:
done should always be called with nothing (OK) or an error, so this line should be

`done(new Error('Expecting FunctionBodyError here'));`

Reply:
changed.